power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/observability.hpp

```diff
@@ -57,11 +59,18 @@ std::vector<int8_t> count_flow_sensors(MeasuredValues<sym> const& measured_value
                     if ((measured_values.has_branch_from(branch) || measured_values.has_branch_to(branch)) &&
                         topo.branch_bus_idx[branch][0] != -1 && topo.branch_bus_idx[branch][1] != -1) {
                         flow_sensors[ybus_index] = 1;
+                        has_at_least_one_sensor = true;
                         break;
                     }
                 }
             }
         }
+        // check voltage sensor
+        if (measured_values.has_voltage(row) && measured_values.has_angle_measurement(row)) {
+            has_at_least_one_sensor = true;
+        }
+        // the system could be ill-conditioned if there is no flow sensor for one bus, except the last bus
+        possibly_ill_conditioned = possibly_ill_conditioned || (!has_at_least_one_sensor && row != n_bus - 1);
     }
```

Comment:
```suggestion
        if (possibly_ill_conditioned) {
            continue;
        }
        // check voltage sensor
        if (measured_values.has_voltage(row) && measured_values.has_angle_measurement(row)) {
            has_at_least_one_sensor = true;
        }
        // the system could be ill-conditioned if there is no flow sensor for one bus, except the last bus
        possibly_ill_conditioned = !has_at_least_one_sensor && row != n_bus - 1;
    }
```
Since `flow_sensors` is not touched and early out is possible.